### PR TITLE
feat(navigation): remove prefix nav for menu and menu-item

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #

## Summary

-  This PR will rename  `nav-menu` to `menu` and `nav-menu-item` to `menu-item`.
-  No longer autoFocus first & last items on ArrowUp & ArrowDown respectively in horizontal layout. 
-  Renamed `subMenuOpen` to `open`  in menu-item.
- Refactor keyboard support . All the interactions are handled in parent calcite-menu.

### CODE REVIEW

File: src/components/menu/menu.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { hidden, renders } from "../../tests/commonTests";
+
+describe("calcite-menu", () => {
+  it("renders", async () => renders("calcite-menu", { display: "flex" }));
+
+  it("honors hidden attribute", async () => hidden("calcite-menu"));
```
Comment (macandcheese) [inline src/components/menu/menu.e2e.ts:6]: Do we want to try adding a11y tests at this point or wait for that?

File: src/components/menu/menu.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ export class CalciteNavMenu {
   @Prop() label: string;
 
   // todo evaluate slotted content and determine if it is a nav menu item, then limit # rendered when auto-collapsing based on width of parent
-  @Prop({ mutable: true }) minCollapsedItems?;
+  @Prop() minCollapsedItems: boolean;
```
Comment (macandcheese) [inline src/components/menu/menu.tsx:39]: We can remove this, for the time being, until we add back the auto-collapse functionality?

File: src/components/nav-menu-item/menu-item.tsx [hunk 3]
```diff
@@ -77,15 +77,15 @@ export class CalciteNavMenuItem implements LoadableComponent {
...
 
   /** Displays the `text` */
-  @Prop({ mutable: true }) textEnabled: boolean;
+  @Prop() textEnabled: boolean;
 
   /**
    * @internal
    */
-  @Prop({ mutable: true }) subMenuOpen = false;
+  @Prop({ mutable: true }) open = false;
```
Comment (macandcheese) [inline src/components/nav-menu-item/menu-item.tsx:88]: I know this is internal - 2 questions - is `open` descriptive enough? And, should this be publicly documented (is there a reason a dev user might want to be able to toggle this? I could see it maybe being useful in `layout="vertical"` menu use cases.

**Review by macandcheese — APPROVED** [review]
Awesome! I think this is good to merge into the wip navigation branch. Some questions, can address once merged.

File: src/components/nav-menu-item/menu-item.tsx (hunk 3 above)
Comment (anveshmekala) [reply src/components/nav-menu-item/menu-item.tsx:88]: i gave a thought about changing it to `open`, the reason being consistency. Though it is internal, keeping it consistent will be useful if we choose to make it public in future as you mentioned for `layout="vertical"`

File: src/components/menu/menu.tsx (hunk 2 above)
Comment (anveshmekala) [reply src/components/menu/menu.tsx:39]: sure, removed.

File: src/components/menu/menu.e2e.ts (hunk 1 above)
Comment (anveshmekala) [reply src/components/menu/menu.e2e.ts:6]: added a11y for `menu-item` and `menu` components.

**Pull request merged**